charts/gardener-extension-shoot-oidc-service/templates/service.yaml

```diff
@@ -0,0 +1,26 @@
+# SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: gardener-extension-{{ .Values.serviceName }}
+  namespace: {{ .Release.Namespace }}
+{{- if .Values.ignoreResources }}
+  annotations:
+    resources.gardener.cloud/ignore: "true"
+{{- end }}
+  labels:
+    app.kubernetes.io/name: gardener-extension-{{ .Values.serviceName }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  type: ClusterIP
+  selector:
+    app.kubernetes.io/name: gardener-extension-{{ .Values.serviceName }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  ports:
+  - port: {{ .Values.webhookConfig.servicePort }}
+    protocol: TCP
+    targetPort: {{ .Values.webhookConfig.serverPort }}
```

Comment:
Could you clarify what is the difference of service and server port? 
In the deployment I see
```yaml
--webhook-config-service-port={{ .Values.webhookConfig.servicePort }}
--webhook-config-server-port={{ .Values.webhookConfig.serverPort }}
```

Which also confused me, but now I see that actually the one port is mapped to the other via the k8s Service.

Reply:
The service port is used by the extension framework to register a webhookconfiguration see [here](https://github.com/gardener/gardener/blob/d19723a97065c6f626d1b9b59b458f71d2cce644/extensions/pkg/webhook/cmd/options.go#L244). On the other hand if not specified it is defaulted to the server port see [here](https://github.com/gardener/gardener/blob/d19723a97065c6f626d1b9b59b458f71d2cce644/extensions/pkg/webhook/cmd/options.go#L227). Maybe it makes sense to remove the `--webhook-config-service-port` configuration and leave it to be defaulted by the extension framework. WDYT?

Reply:
Thanks! Let's keep as it is for now.